framework/helpers/BaseInflector.php

```diff
@@ -383,7 +384,7 @@ public static function camel2id($name, $separator = '-', $strict = false)
         if ($separator === '_') {
             return trim(strtolower(preg_replace($regex, '_\0', $name)), '_');
         } else {
-            return trim(strtolower(str_replace('_', $separator, preg_replace($regex, $separator . '\0', $name))), $separator);
+            return trim(strtolower(preg_replace('/\/'.$separator.'/','/',str_replace('_', $separator, preg_replace($regex, $separator . '\0', $name)))), $separator);
```

Comment:
this function should not be changed. all code necessary to support subfolders in console should be made in yii\console\* classes.

Reply:
Unfortunately there is no special place for only console helpers. Of course this can be done, but making such exception for only one console controller doesn't seem reasonable.

Reply:
handling of path elements does not belong to Inflector, if its needed it needs to be done outside. There is no need to create a helper from it, a private method in the class that is using it is fine.